### DISCUSSION TITLE
v2.11.1

### DISCUSSION THREAD

**Discussion by fiftin** [semaphoreui/semaphore#2658] [Announcements]
## Features
* **Web interface improvements**: UI has received its most significant overhaul in years, making it more polished and professional.
* **Git commit messages**: Tasks now store the Git commit hash and message, enabling you to rerun tasks using the same commit.
* **Task run details**: A new Details page has been added to provide insights into task running results.
* **Terraform/OpenTofu support improvements**: There are many Terraform/OpenTofu support improvements in this release.

## Bugfixes

- **OpenTofu/Terraform**: Fixed a bug with Terraform tasks: the `-auto-approve` flag now works.
- **Security**: Prevented JavaScript from using Cookies to avoid the HTTPOnly issue.
- **OpenTofu/Terraform**: enabled the `-migrate-state` flag by default.
- **OpenTofu/Terraform**: Added prompts for state actions on the web.
- **Terraform/OpenTofu**: Ignored the workspace name for the HTTP backend.

[Semaphore UI 2.11 Release Notes](https://semaphoreui.com/install/docker/2_11/)

## Links

* Create best docker container

   [![](https://img.shields.io/badge/docker_configurator-0050ab?style=for-the-badge&logo=docker)](https://semaphoreui.com/install/docker/2_11/)  

* Join our responsive community

   [![discord](https://img.shields.io/badge/discord_community-510b80?style=for-the-badge&logo=discord)](https://discord.gg/5R6k7hNGcH)  

* Every day we add new features, fix bugs, support the community. We need your support

   [![sponsor](https://img.shields.io/badge/become_a_sponsor-d9c7ff?style=for-the-badge&logo=github-sponsors)](https://github.com/sponsors/semaphoreui) [![ko-fi](https://img.shields.io/badge/buy_me_a_coffee-0ba0e0?style=for-the-badge&logo=kofi)](https://ko-fi.com/fiftin) [![patreon](https://img.shields.io/badge/become_a_patreon-teal?style=for-the-badge&logo=patreon)](https://www.patreon.com/semaphoreui)

<hr /><em>This discussion was created from the release <a href='https://github.com/semaphoreui/semaphore/releases/tag/v2.11.1'>v2.11.1</a>.</em>

**Comment by cmennens** [comment]
UI improvement is amazing! Great work & thank you!
